### DISCUSSION TITLE
"The printer may not exist or is unavailable at this time."

### DISCUSSION THREAD

**Discussion by Laanan** [OpenPrinting/cups#338] [Q&A] [answered]
When I try to print I get the error: "The printer may not exist or is unavailable at this time."

The job will just stay stuck in "processing" forever. This includes test prints. All the settings appear normal in /etc/cups/cups-files.conf, as well as on the web interface. I am using the cups recommended driver for the printer (an HPmfp428f). Restarting the computer and restarting the cups server does nothing. I have installed the backend cups packages as well. This is for a vanilla Deb 11 system, newly set-up.

Would someone please suggest further troubleshooting tips?

**Comment by zdohnal** [comment]
Hi @Laanan ,

the error happens when you have problems with connecting to the printer - debugging it depends on what connection (usb/network) and queue type (temporary/permanent) do you use for your printer.  For usb you can check output of ```lsusb -v```, network printers can be checked by pinging ip/hostname, using telnet to check whether your desired port is opened on the printer, for temporary queues check Avahi messages as well (via ```avahi-browse -avrt```). And using wireshark/tcpdump to capture the packet communication (works for USB and network as well) can tell you more information.

In the end don't forget to clean up your queue during debugging - you can clog the queue with one job and then next job gets queued. This depends on your error policy in cupsd.conf as well.

**Comment by Laanan** [comment, marked as answer]
Hi @zdohnal,

Thanks so much for the response. You were completely correct. My printer has a direct connect wifi, so I connected to it and was able to print a test page.

Turns out it was an issue with my Sonicwall. I had a rule allowing WAN to LAN for all users and all subnets. Oddly, to be able to print I had to make a more specific rule allowing WAN to LAN on my specific subnet. Really does not make sense to me why this setting change worked, but I actually had a similar issue in the last with an SSL secured VPN client. 

So, nothing to do with CUPS whatsoever, but thanks for pointing me in the right direction! 👍🏻
